src/core/Akka.Remote/Transport/DotNetty/BatchWriter.cs

```diff
@@ -134,52 +144,50 @@ public override Task WriteAsync(IChannelHandlerContext context, object message)
             return write;
         }
 
+        public override void Flush(IChannelHandlerContext context)
+        {
+            // reset statistics upon flush
+            Reset();
+            base.Flush(context);
+        }
+
+        public override void UserEventTriggered(IChannelHandlerContext context, object evt)
+        {
+            if (evt is FlushCmd)
+            {
+                if (HasPendingWrites)
+                {
+                    context.Flush();
+                    Reset();
+                }
+            }
+        }
+
         public override Task CloseAsync(IChannelHandlerContext context)
         {
             // flush any pending writes first
             context.Flush();
+            _flushSchedule?.Cancel();
             CanSchedule = false;
             return base.CloseAsync(context);
         }
 
         private void ScheduleFlush(IChannelHandlerContext context)
         {
             // Schedule a recurring flush - only fires when there's writable data
-            var task = new FlushTask(context, Settings.FlushInterval, this);
-            context.Executor.Schedule(task, Settings.FlushInterval);
+            _flushSchedule = Scheduler.Advanced.ScheduleRepeatedlyCancelable(Settings.FlushInterval,
+                Settings.FlushInterval,
+                () =>
+                {
+                    // want to fire this event through the top of the pipeline
+                    context.Channel.Pipeline.FireUserEventTriggered(FlushCmd.Instance);
+                });
```

Comment:
So this is one heck of an edge case.... But do we want to have any sort of warning if the flush interval is for some unknown reason less than `tick-duration` on HWT?

Reply:
No we don't - but it shouldn't be an issue. The scheduler will just put it in the next available bucket if that's the case.